api/src/main/java/io/druid/data/input/impl/StringDimensionSchema.java

```diff
@@ -27,15 +27,19 @@
 {
   @JsonCreator
   public static StringDimensionSchema create(String name) {
-    return new StringDimensionSchema(name);
+    return new StringDimensionSchema(name, -1);
   }
 
+  private final int compareCacheSize;
+
   @JsonCreator
   public StringDimensionSchema(
-      @JsonProperty("name") String name
+      @JsonProperty("name") String name,
+      @JsonProperty("compareCacheSize") int compareCacheSize
```

Comment:
I wonder if this may also cause issues when merging segments where one segment may have a cache and the other one doesn't.